gr-uhd/python/uhd/replaymsgpushbutton.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2023 Ettus Research, a National Instruments Brand.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+#
+
+from PyQt5 import Qt
+from gnuradio import gr
+import pmt
+
+
+class ReplayMsgPushButton(gr.sync_block, Qt.QPushButton):
+    """
+    This block creates a variable push button that creates a message
+    when clicked. The message will be formatted as a dictionary to pass
+    to the RFNoC Replay block
+    """
+
+    def __init__(self, lbl, relBackColor, relFontColor,
+                 command, port, offset=-1, size=-1, time=-1):
+        gr.sync_block.__init__(self, name="ReplayMsgPushButton",
+                               in_sig=None, out_sig=None)
+        Qt.QPushButton.__init__(self, lbl)
+
+        self.lbl = lbl
+
+        self.replayDict = pmt.dict_add(pmt.make_dict(), pmt.intern('command'), pmt.intern(command))
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('port'), pmt.from_long(port))
+        # If the user does not specify a command offset, size, or time, don't add to the command.
+        # The properties will be used for offset and size, and the command will happen immediately.
+        if offset != -1:
+            self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('offset'), pmt.from_long(offset))
+        if size != -1:
+            self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('size'), pmt.from_long(size))
+        if time != -1:
+            self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('time'), pmt.from_float(time))
+
+        styleStr = ""
+        if relBackColor != 'default':
+            styleStr = "background-color: " + relBackColor + "; "
+
+        if relFontColor:
+            styleStr += "color: " + relFontColor + "; "
+
+        self.setStyleSheet(styleStr)
+
+        self.clicked[bool].connect(self.onBtnClicked)
+
+        self.message_port_register_out(pmt.intern("pressed"))
+
+    def set_command(self, command):
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('command'), pmt.intern(command))
+
+    def set_port(self, port):
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('port'), pmt.from_long(port))
+
+    def set_size(self, size):
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('size'), pmt.from_long(size))
+
+    def set_offset(self, offset):
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('offset'), pmt.from_long(offset))
+
+    def set_time(self, time):
+        self.replayDict = pmt.dict_add(self.replayDict, pmt.intern('time'), pmt.from_float(time))
+
+    def onBtnClicked(self, pressed):
+        self.message_port_pub(pmt.intern("pressed"), self.replayDict)
```

Comment:
If you want to make it a Python dict, then you need to change this line to
```suggestion
        self.message_port_pub(pmt.intern("pressed"), pmt.to_pmt(self.replayDict))
```